api/client/proto/proto.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package proto
+
+import (
+	"github.com/gravitational/teleport/api/types"
+
+	"github.com/gravitational/trace"
+)
+
+// FromWatchKind converts the watch kind value between internal
+// and the protobuf format
+func FromWatchKind(wk types.WatchKind) WatchKind {
+	return WatchKind{
+		Name:        wk.Name,
+		Kind:        wk.Kind,
+		SubKind:     wk.SubKind,
+		LoadSecrets: wk.LoadSecrets,
+		Filter:      wk.Filter,
+	}
+}
+
+// ToWatchKind converts the watch kind value between the protobuf
+// and the internal format
+func ToWatchKind(wk WatchKind) types.WatchKind {
+	return types.WatchKind{
+		Name:        wk.Name,
+		Kind:        wk.Kind,
+		SubKind:     wk.SubKind,
+		LoadSecrets: wk.LoadSecrets,
+		Filter:      wk.Filter,
+	}
+}
```

Comment:
This is good question. I've had a similar concern with @Joerger api refactoring [PR](https://github.com/gravitational/teleport/pull/5384#discussion_r564455486) - the stand there is that the api client needs to manage a separate set of types unrelated to protobuf to avoid leaking implementation details about internal use of gRPC which I have trouble justifying.
If, on the hand, the request types were to be removed from the protobuf side, it would look equally strange. So some consensus is required.

Reply:
The way we deduplicate this currently is to define the type in `api/types/types.proto`. You can then add methods in `types/sessions.go`, and import the types into `authservice.proto`. You can look at types like `types.CertAuthorityV2` for an example of that. 

But the problem here is that WatchKind is defined in `authservice.proto`. It should probably be deprecated and moved to `types.proto` to solve this issue. This is also an issue with `Events` and `Duration` which should likewise be refactored eventually.

Reply:
I'll deduplicate in a follow-up PR.